library/core/src/ptr/non_null.rs

```diff
@@ -1177,6 +1185,37 @@ impl<T: ?Sized> NonNull<T> {
     #[must_use]
     #[stable(feature = "non_null_convenience", since = "1.80.0")]
     #[rustc_const_unstable(feature = "const_align_offset", issue = "90962")]
+    #[requires(!self.pointer.is_null())]
+    #[ensures(|result| {
+        // Post-condition reference: https://github.com/model-checking/verify-rust-std/pull/69/files
```

Comment:
It seems a good reference, maybe we could introduce this to other team mates in the coding session